c7n/resources/glue.py

```diff
@@ -165,6 +165,54 @@ def process(self, resources):
                 continue
 
 
+@GlueJob.action_registry.register('enable-metrics')
+class GlueJobEnableMetrics(BaseAction):
+    """Enable CloudWatch metrics for a Glue job
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gluejob-enable-metrics
+            resource: glue-job
+            filters:
+              - type: value
+                key: 'DefaultArguments."--enable-metrics"'
+                value: absent
+            actions:
+              - type: enable-metrics
+    """
+    schema = type_schema('enable-metrics')
+    permissions = ('glue:UpdateJob',)
+
+    def cleanup_params(self, r):
```

Comment:
I see the cleanup here is necessary, but it looks like as-is this will mutate the existing resource. That could trip up subsequent actions, affect resources.json output or die altogether trying to remove `Tags` if a job is untagged.

Another option here might be something like:

```python
params = {
    k: r[k] for k  in
    set(r.keys()).intersection(client.meta._service_model.shape_for('JobUpdate').members)
}
```

So rather than picking out keys to delete from the original resource, we'd extract the ones that are components of a `JobUpdate` into a new object, continue with the `MaxCapacity` check/fix and return from that.

Reply:
@ajkerrigan ty for the tip on how to better handle the param clean up. I'll try that out. As for disabling metrics, I can just update the action to be `toggle-enable-metrics` and let it take in a property called `enabled`. Based on that, we can either enable or disable metrics collection for the glue job. How does that sound?

Reply:
Updated